catwalk/models/metaicl.py

```diff
@@ -0,0 +1,133 @@
+                else:
+                    fewshot_instance['input'] =  self._apply_meta_icl_per_instance_truncation(fewshot_instance, tokenizer)
+                truncated_fewshot_instances.append(fewshot_instance)
+            
+            rc_instances.append(task.convert_instance(
+                instance,
+                InstanceFormat.RANK_CLASSIFICATION,
+                fewshot_instances=truncated_fewshot_instances,
+                continuation_seperator=self.continuation_seperator,
+                example_seperator=self.example_seperatior,
+            ))
+        
+        rc_instances = [self._apply_meta_icl_concatinated_input_truncation(instance, tokenizer) for instance in rc_instances]
+
+        # get all the tuples
+        for instance_index, rc_instance in enumerate(rc_instances):
+            for instance_request in rc_instance.choices:
+                instance_index_to_tuple_indices[instance_index].append(len(tuples))
+                tuples.append(instance_request)
+
+        # run the requests
+        results = self._run_loglikelihood(tuples, model, tokenizer, batch_size)
+
+        # TODO figure out why this is being suppressed
+        if self.instances_truncated == self.instances_total:
+            warnings.warn("All examples in this dataset chunk are being truncated after concatenation, consider using smaller max_length_per_example")
+
+        # collect the results
+        for instance_index, rc_instance in enumerate(rc_instances):
+            tuple_indices = instance_index_to_tuple_indices[instance_index]
+            results_for_instance = [results[i] for i in tuple_indices]
+            result_tensor = torch.tensor(results_for_instance)
+            metric_args = (result_tensor, rc_instance.correct_choice)
+            yield {
+                "acc": metric_args,
+                "f1": metric_args,
+                "precision": metric_args,
+                "recall": metric_args,
+            }
+        
+    def _apply_meta_icl_per_instance_truncation(self, instance: Dict[str, Any], tokenizer: _Tokenizer, is_icl_demonstration: bool = True, is_first: bool = False) -> str:
+        """Applies identical truncation as in metaicl.data.py.MetaICLData._prepro_each_datapoint"""
+        # formatting is added and then thrown away just because it is needed for truncation lengths
+        input_tokens = tokenizer(('' if is_first else self.example_seperatior) + instance['input'])["input_ids"]
+
+        if is_icl_demonstration:
+            output_tokens = tokenizer(self.continuation_seperator + instance["output"])["input_ids"]
+            input_tokens = input_tokens[:self.max_length_per_example - 2 - len(output_tokens)]
+            assert len(input_tokens)+len(output_tokens)+2<=self.max_length_per_example, \
+                (instance.get("task", None), len(input_tokens), len(output_tokens), self.max_length_per_example)
```

Comment:
Is this for debugging?

Reply:
The assert? It's from the MetaICL code. I agree it's unnecessary. I'll remove it.